lib/StripeResource.js

```diff
@@ -192,14 +197,50 @@ StripeResource.prototype = {
       callback.call(
         self,
         new Error.StripeConnectionError({
-          message: 'An error occurred with our connection to Stripe',
+          message: self._generateConnectionErrorMessage(requestRetries),
           detail: error,
         }),
         null
       );
     }
   },
 
+  _shouldRetry: function(res, numRetries) {
+    // Retry on connection error, if we have retries left.
+    if (!res) {
+      return numRetries < this._stripe.getMaxNetworkRetries();
+    }
+
+    // Retry on conflict or rate limit error, if we have retries left.
+    if (res.statusCode === 409 || res.statusCode === 429) {
+      return numRetries < this._stripe.getMaxNetworkRetries();
```

Comment:
quick question:
the two errors we encounter most often when calling stripe that would be helped by using retries are 
- StripeAPIError
- StripeConnectionError

I cant find the http codes for those two, but wondering if this list would cover these two errors. Might be nice to let the client set the statuscodes to retry? i wonder if that'd open up the gates to people starting to retry 400s etc

Reply:
Hi @suchitagarwal , thanks, that's very helpful to hear!

As it is, the code will _not_ handle `StripeAPIError`, which is status code 500, but _will_ handle `StripeConnectionError`. I'm going to change that to handle 500's too, but only when they are GET's or DELETE's (not POST). 

The reason for this is that there isn't really much point to retrying a 500'd POST request with an idempotency key, because our idempotency machinery will simply give you a 500 again.

I don't think we'll open up the logic for when to automatically retry, pretty much based for reasons like you mentioned (retrying a 400 isn't a great idea), but users can always implement retries themselves. 

Thoughts?

Reply:
makes perfect sense. i think handling 500s for GET/DELETE should cover the cases we encounter most often, and not retrying POSTs makes sense. thanks a lot for the quick turnaround on this

Reply:
Awesome, thanks for the feedback!

Speaking of, I'm working on an unrelated feature and looking for some early beta users... if you're willing to try something out, drop me a line? email is `githubUsername.replace('-', '@') + '.com'` 😉